Lib/test/eintrdata/eintr_tester.py

```diff
@@ -481,13 +482,45 @@ def test_devpoll(self):
         self.assertGreaterEqual(dt, self.sleep_time)
 
 
+class FNTLEINTRTest(EINTRBaseTest):
+    def _lock(self, lock_func, lock_name):
+        code = '\n'.join((
+            "import fcntl, time",
+            "with open('%s', 'wb') as f:" % support.TESTFN,
+            "   fcntl.%s(f, fcntl.LOCK_EX)" % lock_name,
+            "   time.sleep(%s)" % self.sleep_time))
+        proc = self.subprocess(code)
+        with open(support.TESTFN, 'wb') as f:
+            while True:  # synchronize the subprocess
+                try:
+                    lock_func(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                    lock_func(f, fcntl.LOCK_UN)
+                except BlockingIOError:
+                    break
```

Comment:
In what way? As far I understand the exception could be raised only if the other process is in the right state (managed to acquire the lock). I do see that it may use yield, after unlocking to improve performance.

Reply:
I'm sorry, I fail to see why this loop seems to be unreliable to you, could you give me a bit more details? There are two ways to get out of it, one through interruption and one because of BlockingIOError. The first one is test failure anyway, so it can be ignored, the other one means that the subprocess managed to acquire the lock so we can start the real testing.